packages/utils/src/lib/transform.ts

```diff
@@ -156,3 +157,56 @@ export function toOrdinal(value: number): string {
   return `${value}th`;
 }
 /* eslint-enable no-magic-numbers */
+
+export class AuditsNotImplementedError extends Error {
+  constructor(list: { slug: string }[], auditSlugs: string[]) {
+    super(
+      `audits: "${auditSlugs
+        .filter(slug => !list.some(a => a.slug === slug))
+        .join(', ')}" not implemented`,
+    );
+  }
+}
+
+export function filterByAuditSlug<
+  T extends {
+    refs: GroupRef[];
+  },
+  L extends T[],
+  S extends L[number]['refs'][number]['slug'],
+>(groups: L, auditSlugs: S | S[]): L {
+  const slugs = toArray(auditSlugs);
+  if (slugs.length === 0) {
+    return groups;
+  }
+  return (
+    groups
+      // filter out groups that have no audits includes from onlyAudits (avoid empty groups)
+      .filter(group => group.refs.some(({ slug }) => slugs.includes(slug as S)))
+      .map(group => {
+        const groupsRefs = group.refs.filter(({ slug }) =>
+          slugs.includes(slug as S),
+        );
+
+        return {
+          ...group,
+          refs: groupsRefs,
+        };
+      }) as L
+  );
+}
+
+export function filterBySlug<T extends Audit, L extends T[]>(
```

Comment:
Same as `filterByAuditSlug`. Please use verbose names for both types and variables and reduce the complexity of the types if possible.

Reply:
Not relevant anymore I believe, as mentioned elsewhere. But in general a good point, when there are multiple type parameters, then `TAudit`, `TList` etc. is often a better naming convention.